pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/util/ValueReader.java

```diff
@@ -63,7 +64,7 @@ default byte[] getUnpaddedBytes(int index, int numBytesPerValue, byte[] buffer)
    */
   default String getUnpaddedString(int index, int numBytesPerValue, byte[] buffer) {
     int length = readUnpaddedBytes(index, numBytesPerValue, buffer);
-    return new String(buffer, 0, length);
+    return new String(buffer, 0, length, StandardCharsets.UTF_8);
```

Comment:
(not blocking)

String in most JDKs support LATIN1 and UTF-16. My understanding is that most Latin/English character strings end up using LATIN1 which is reasonably compact (e.g. UUID is 36 characters).

Would passing UTF_8 switch the coder value to UTF-16 for some other strings that could have been served with Latin-1? It could lead to a quite a big increasing in heap utilization for such cases, since UTF-16 is almost double the size of Latin-1.

<img width="753" height="265" alt="image" src="https://github.com/user-attachments/assets/03ab8977-a2c9-425f-b249-b4a5c4c33a20" />

Reply:
My understanding is UTF_8 decoding should not force UTF-16 storage, and how Java stores it only depends on the characters in the string. 
<img width="646" height="322" alt="Screenshot 2025-08-15 at 1 27 54 PM" src="https://github.com/user-attachments/assets/cbe69490-abc3-4df5-af2b-93f682386e93" />